docs/spec/spec.md

```diff
@@ -267,6 +273,189 @@ The GraphQL engine acts as the main processing unit in the Ballerina GraphQL pac
 
+### 2.5 Client
+
+The GraphQL client can be used to connect to a GraphQL service and retrieve data. This client currently supports the `Query` and `Mutation` operations. The Ballerina GraphQL client uses HTTP as the underlying protocol to communicate with the GraphQL service.
+
+#### 2.5.1 Client Configuration
+
+The `graphql:Client` uses `http:Client` as its underlying implementation; this `http:Client` can be configured by providing the `graphql:ClientConfiguaration` as an optional parameter via `graphql:Client` init method.
+
+```ballerina
+# Provides a set of configurations for controlling the behavior of the GraphQL client when communicating with 
+# the GraphQL server that operates over HTTP.
+#
+# + http1Settings - Configurations related to HTTP/1.1 protocol
+# + timeout - The maximum time to wait (in seconds) for a response before closing the connection
+# + forwarded - The choice of setting `forwarded`/`x-forwarded` header
+# + followRedirects - Configurations associated with Redirection
+# + poolConfig - Configurations associated with request pooling
+# + cache - HTTP caching related configurations
+# + compression - Specifies the way of handling compression (`accept-encoding`) header
+# + auth - Configurations related to client authentication
+# + circuitBreaker - Configurations associated with the behavior of the Circuit Breaker
+# + retryConfig - Configurations associated with retrying
+# + cookieConfig - Configurations associated with cookies
+# + responseLimits - Configurations associated with inbound response size limits
+# + secureSocket - SSL/TLS-related options
+# + proxy - Proxy server related options
+# + validation - Enables the inbound payload validation functionality which is provided by the constraint package. Enabled by default
+public type ClientConfiguration record {|
+    ClientHttp1Settings http1Settings = {};
+    decimal timeout = 60;
+    string forwarded = "disable";
+    FollowRedirects? followRedirects = ();
+    PoolConfiguration? poolConfig = ();
+    CacheConfig cache = {};
+    Compression compression = COMPRESSION_AUTO;
+    ClientAuthConfig? auth = ();
+    CircuitBreakerConfig? circuitBreaker = ();
+    RetryConfig? retryConfig = ();
+    CookieConfig? cookieConfig = ();
+    ResponseLimitConfigs responseLimits = {};
+    ClientSecureSocket? secureSocket = ();
+    ProxyConfig? proxy = ();
+    boolean validation = true;
+|};
+```
+
+#### 2.5.2 Initializing the Client
+
+The `graphql:Client` init method requires a valid URL and optional configuration to initialize the client. 
+
+```ballerina
+graphql:Client graphqlClient = check new (“http://localhost:9091/graphql”, {timeout: 10});
```

Comment:
Let's use the port `4000` for the sample GraphQL services to maintain consistency. (Please check whether there are any other places used a port other that `4000` and update them).

Reply:
Updated